src/Annotator/index.js

```diff
@@ -102,19 +104,30 @@ export const Annotator = ({
   )
 
   const dispatch = useEventCallback((action: Action) => {
-    if (
-      action.type === "HEADER_BUTTON_CLICKED" &&
-      ["Exit", "Done", "Save", "Complete"].includes(action.buttonName)
-    ) {
-      onExit(without(state, "history"))
-    } else {
-      dispatchToReducer(action)
+    if (action.type === "HEADER_BUTTON_CLICKED") {
+      if (["Exit", "Done", "Save", "Complete"].includes(action.buttonName)) {
+        return onExit(without(state, "history"))
+      } else if (action.buttonName === "Next") {
+        return onNextImage(without(state, "history"))
+      } else if (action.buttonName === "Prev") {
+        return onPrevImage(without(state, "history"))
+      }
     }
```

Comment:
intercept action.type === "NEXT_IMAGE"  and call onNextImage if it's defined. Same for onPrevImage

condition should be changed to `if (action.buttonName === "Next" && onNextImage) {`